### ISSUE TITLE
error: illegal IP address string passed to inet_pton

### ISSUE THREAD

**Issue by lucas03** [Miserlou/django-easy-timezones#6]
I get this error when bots are trying to access my django site. Any idea how illegal IP address should be dealt with?
Internal Server Error: /
Traceback (most recent call last):
  File "/var/lib/openshift/5409d566e0b8cd8c7a0002b6/python/virtenv/lib/python2.7/site-packages/django/core/handlers/base.py", line 87, in get_response
    response = middleware_method(request)
  File "/var/lib/openshift/5409d566e0b8cd8c7a0002b6/python/virtenv/lib/python2.7/site-packages/easy_timezones/middleware.py", line 42, in process_request
    tz = db.time_zone_by_addr(ip)
  File "/var/lib/openshift/5409d566e0b8cd8c7a0002b6/python/virtenv/lib/python2.7/site-packages/pygeoip/__init__.py", line 583, in time_zone_by_addr
    ipnum = util.ip2long(addr)
  File "/var/lib/openshift/5409d566e0b8cd8c7a0002b6/python/virtenv/lib/python2.7/site-packages/pygeoip/util.py", line 39, in ip2long
    return int(binascii.hexlify(socket.inet_pton(socket.AF_INET6, ip)), 16)
error: illegal IP address string passed to inet_pton

**Comment by jasonm** [comment]
@vinitkumar's patch works for me.  I was encountering this in the https://github.com/torchbox/wagtail CMS admin panel when previewing a page.

In case anyone is following along at home...

Wagtail's "preview this page" functionality issues a dummy request to compute the previewed content.  This presumably passes a fake requestor IP address into the dummy request.  The django-easy-timezones library has a piece of middleware installed which attempts to infer a time zone based on geolocating the requestor IP.  So, when Wagtail issues the dummy request, the bad IP address causes django-easy-timezone to pass a malformed IP to inet_aton, which blows up.

@vinitkumar's additional well-formedness check will defensively prevent not only IPV6 addresses from causing an issue, but any malformed IP address.

**Comment by jasonm** [comment]
Looking more closely at `easy_timezones.middleware.EazyTimezoneMiddleware.process_request`, is it reasonable for it to change:

```python
            if ip != '127.0.0.1':
                # if not local, fetch the timezone from pygeoip
                tz = db.time_zone_by_addr(ip)
```

to also check for a missing IP:

```python
            if ip and ip != '127.0.0.1':
                # if not local, fetch the timezone from pygeoip
                tz = db.time_zone_by_addr(ip)
```

?

**Comment by jasonm** [comment]
There seems to be some prior Django discussion that discourages relying on `REMOTE_ADDR`: https://www.djangoproject.com/weblog/2009/jul/28/security/#secondary-issue so perhaps the `if ip and ip != '127.0.0.1'` is the way to go?

**Comment by Miserlou** [comment]
I like that @jasonm.

PR's graciously accepted - if you're using the IPV6 database, please update the documentation to reflect that as well.

Thanks!

**Comment by Miserlou** [comment]
The supplied code wasn't totally valid, but I've added a different solution and now it should work. I've also started adding tests.

**Comment by lucas03** [comment]
Nice! will you push it to pip as well? Will you add IPv6 support with GEOIP file for IPv6?

**Comment by Miserlou** [comment]
@lucas03 - IPv6 support based on the GeoIPv6 file is available in version 0.6.0, published to pip.

Both of the GeoIP files are now included by default for convenience.
